src/main/java/core/basesyntax/FieldXY.java

```diff
@@ -13,7 +13,42 @@
 public class FieldXY {
 
     public void moveRobot(Robot robot, int toX, int toY) {
-        //your code
+        while ((robot.getX() != toX) || (robot.getY() != toY)) {
+            goToX(robot,toX);
+            goToY(robot,toY);
+        }
+    }
+
+    public static void goToX(Robot robot, int newX) {
+        if (robot.getX() < newX) {
+            while (robot.getDirection() != Direction.RIGHT) {
+                robot.turnRight();
+            }
+        }
+        if (robot.getX() > newX) {
+            while (robot.getDirection() != Direction.LEFT) {
+                robot.turnRight();
+            }
+        }
+        while (robot.getX() != newX) {
+            robot.stepForward();
+        }
+    }
```

Comment:
each method should have unique logic. What is the difference in logic between this method and goToY?

Reply:
redid by one method and added constants